### PR TITLE
[WC-2341]: Change Combo Box to allow for Ranked Sorting again

### PR DESCRIPTION
**Pull request by samuelreichert**
### Pull request type

Bug fix (non-breaking change which fixes an issue)

---

### Description

The idea is to return the ranked sorting again to combobox, but only when user starts typing something.

### CODE REVIEW

File: packages/pluggableWidgets/combobox-web/src/helpers/Database/DatabaseOptionsProvider.ts [hunk 1]
```diff
@@ -1,21 +1,19 @@
-import { CaptionsProvider, OptionsProvider, Status } from "../types";
+import { CaptionsProvider, Status } from "../types";
 import { ListValue, ObjectItem } from "mendix";
-import { matchSorter } from "match-sorter";
 import { FilterTypeEnum } from "../../../typings/ComboboxProps";
+import { BaseOptionsProvider } from "../BaseOptionsProvider";
 
 interface Props {
```
Comment (gjulivan) [inline packages/pluggableWidgets/combobox-web/src/helpers/Database/DatabaseOptionsProvider.ts:6]: i think we can move this interface to base options provider, and export from there.

File: packages/pluggableWidgets/combobox-web/src/helpers/Association/AssociationOptionsProvider.ts [hunk 2]
```diff
@@ -1,22 +1,20 @@
-import { CaptionsProvider, OptionsProvider, Status } from "../types";
+import { CaptionsProvider, Status } from "../types";
 import { ListValue, ObjectItem, ReferenceSetValue, ReferenceValue } from "mendix";
-import { matchSorter } from "match-sorter";
 import { FilterTypeEnum } from "../../../typings/ComboboxProps";
+import { BaseOptionsProvider } from "../BaseOptionsProvider";
 
 interface Props {
```
Comment (gjulivan) [inline packages/pluggableWidgets/combobox-web/src/helpers/Association/AssociationOptionsProvider.ts:6]: and this should extend from Props, with added 'attr'

File: packages/pluggableWidgets/combobox-web/src/helpers/BaseOptionsProvider.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import { FilterTypeEnum } from "typings/ComboboxProps";
+import { CaptionsProvider, OptionsProvider, Status } from "./types";
+import { MatchSorterOptions, matchSorter } from "match-sorter";
+import { getFilterTypeOptions } from "./utils";
+
+export class BaseOptionsProvider<T = unknown, P = object> implements OptionsProvider<T, P> {
```
Comment (gjulivan) [inline packages/pluggableWidgets/combobox-web/src/helpers/BaseOptionsProvider.ts:6]: P can't be random object, should be extends interface Props

File: packages/pluggableWidgets/combobox-web/src/helpers/BaseOptionsProvider.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        this.trigger = callback;
+    }
+
+    loadMore(): void {
+        return undefined;
+    }
+
+    _updateProps(_props: P): void {}
+    _optionToValue(_option: string | null): T | undefined {
+        return undefined;
```
Comment (gjulivan) [inline packages/pluggableWidgets/combobox-web/src/helpers/BaseOptionsProvider.ts:54]: then we can actually fill this in base provider

Comment (gjulivan) [inline packages/pluggableWidgets/combobox-web/src/helpers/BaseOptionsProvider.ts:54]: this should throw an exception, not implemented

File: packages/pluggableWidgets/combobox-web/src/helpers/BaseOptionsProvider.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    }
+    _valueToOption(_value: T | undefined): string | null {
+        return null;
```
Comment (gjulivan) [inline packages/pluggableWidgets/combobox-web/src/helpers/BaseOptionsProvider.ts:57]: this should throw an exception not implemented

**Review by gjulivan — APPROVED** [review]

**Comment by gjulivan** [comment]
please add changelog

**Review by gjulivan — APPROVED** [review]

**Review by leonardomendix — APPROVED** [review]

**Pull request merged**